src/plugins/intel_gpu/tests/test_cases/canonicalization_gpu_test.cpp

```diff
@@ -0,0 +1,167 @@
+
+    ov::PartialShape expected_input_pshape{1, 2, 3, 1, 1, 1};
+    EXPECT_TRUE(canonicalized_impl_param.input_layouts[0].get_partial_shape() == expected_input_pshape);
+
+    ov::PartialShape expected_output_pshape{3, 1, 1, 1, 1, 1};
+    EXPECT_TRUE(canonicalized_impl_param.output_layouts[0].get_partial_shape() == expected_output_pshape);
+}
+
+TEST(canonicalization, select) {
+    auto& engine = get_test_engine();
+
+    layout inputs_layout {ov::PartialShape{2, 2}, data_types::f32, format::bfyx};
+
+    topology topology;
+    topology.add(input_layout("mask", inputs_layout));
+    topology.add(input_layout("input1", inputs_layout));
+    topology.add(input_layout("input2", inputs_layout));
+    topology.add(cldnn::select("select", input_info("mask"), input_info("input1"), input_info("input2")));
+
+    ExecutionConfig config;
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(true));
+    network network(engine, topology, config);
+
+    auto inst = network.get_primitive("select");
+    auto impl = inst->get_impl();
+    ASSERT_TRUE(impl != nullptr);
+
+    auto impl_param = inst->get_node().get_kernel_impl_params();
+    auto canonicalized_impl_param = impl->canonicalize_shapes(*impl_param);
+
+    ov::PartialShape expected_input_pshape{1, 1, 2, 2};
+    for (const auto& input_layout : canonicalized_impl_param.input_layouts) {
+        EXPECT_TRUE(input_layout.get_partial_shape() == expected_input_pshape);
+    }
+
+    auto expected_output_pshape = expected_input_pshape;
+    EXPECT_TRUE(canonicalized_impl_param.output_layouts[0].get_partial_shape() == expected_output_pshape);
+}
+
+TEST(canonicalization, broadcast) {
+    auto& engine = get_test_engine();
+
+    layout input0_layout {ov::PartialShape{5}, data_types::f32, format::bfyx};
+
+    topology topology;
+    topology.add(input_layout("input", input0_layout));
+    topology.add(broadcast("broadcast", input_info("input"), ov::Shape{3, 1, 5}, {}, ov::op::BroadcastType::NUMPY));
+
+    ExecutionConfig config;
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(true));
+    network network(engine, topology, config);
+
+    auto inst = network.get_primitive("broadcast");
+    auto impl = inst->get_impl();
+    ASSERT_TRUE(impl != nullptr);
+
+    auto impl_param = inst->get_node().get_kernel_impl_params();
+    auto canonicalized_impl_param = impl->canonicalize_shapes(*impl_param);
+
+    ov::PartialShape expected_input_pshape{1, 1, 5, 1};
+    EXPECT_TRUE(canonicalized_impl_param.input_layouts[0].get_partial_shape() == expected_input_pshape);
+
+    ov::PartialShape expected_output_pshape{3, 1, 5, 1};
+    EXPECT_TRUE(canonicalized_impl_param.output_layouts[0].get_partial_shape() == expected_output_pshape);
+}
+
+TEST(canonicalization, eltwise) {
+    auto& engine = get_test_engine();
+
+    layout input0_layout {ov::PartialShape{2, 2, 3}, data_types::f32, format::bfyx};
+    layout input1_layout {ov::PartialShape{2, 3}, data_types::f32, format::bfyx};
```

Comment:
I think tests should be parameterized to check more shape combinations if needed.